tests/unit/s2n_tls13_pq_handshake_test.c

```diff
@@ -406,6 +406,27 @@ int main()
         .ecc_preferences = &s2n_ecc_preferences_20201021,
     };
 
+    const struct s2n_kem_group *mlkem768_test_groups[] = {
+        &s2n_x25519_mlkem_768,
```

Comment:
If I'm understanding this test correctly, I think this will only try to handshake with `s2n_x25519_mlkem_768`? Would it be worth adding a self-talk test here or somewhere else that negotiates both of these?

Reply:
Yes, that's correct. I have updated s2n_tls13_pq_handshake_test.c to iterate over every KemGroup in `ALL_SUPPORTED_KEM_GROUPS` and perform a handshake with each one individually.